### PR TITLE
Implements generating application secret

### LINKED ISSUE #951: Web for secrets management feature
**What would you like to be added**:

**Why is this needed**:

/label area/web
/assign @cakecatz

### PR DESCRIPTION
**Pull request by cakecatz**
**What this PR does / why we need it**:

Add this feature to the application list menu.

image

image

Kapture 2020-10-20 at 17 05 25

**Which issue(s) this PR fixes**:

Fixes #951  

**Does this PR introduce a user-facing change?**:

```release-note
Add generating application secret at the application list menu
```

### CODE REVIEW

File: pkg/app/web/src/components/application-list.tsx [hunk 1]
```diff
@@ -235,14 +261,23 @@ export const ApplicationList: FC = memo(function ApplicationList() {
         ) : (
           <MenuItem onClick={handleOnClickDisable}>Disable</MenuItem>
         )}
+        <MenuItem onClick={handleOnClickGenerateSecret}>
+          Generate Secret
```
Comment (nghialv) [inline pkg/app/web/src/components/application-list.tsx:265]: `Encrypt Secret`

File: pkg/app/web/src/components/sealed-secret-dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  },
+}));
+
+interface Props {
+  applicationId: string | null;
+  open: boolean;
+  onClose: () => void;
+}
+
+const DIALOG_TITLE = "Generate application sealed secret";
```
Comment (nghialv) [inline pkg/app/web/src/components/sealed-secret-dialog.tsx:37]: "Encrypting secret data for application"

File: pkg/app/web/src/components/sealed-secret-dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+          <DialogActions>
+            <Button onClick={onClose} disabled={sealedSecret.isLoading}>
+              Cancel
+            </Button>
+            <Button
+              color="primary"
+              type="submit"
+              disabled={sealedSecret.isLoading}
+            >
+              Generate
```
Comment (nghialv) [inline pkg/app/web/src/components/sealed-secret-dialog.tsx:138]: "Encrypt"

**Comment by cakecatz** [comment]
@nghialv Thank you.
Updated texts.

**Comment by nghialv** [comment]
Thank you.
/approve

**Pull request merged**